cmd/cli/plugin/package/main.go

```diff
@@ -52,3 +54,22 @@ func getOutputFormat() string {
 	}
 	return format
 }
+
+func isPackagingAPIAvailable(kubeCfgPath string) (bool, error) {
+	cfg, err := kappclient.GetKubeConfig(kubeCfgPath)
+	if err != nil {
+		return false, err
+	}
+	clusterQueryClient, err := capdiscovery.NewClusterQueryClientForConfig(cfg)
+	if err != nil {
+		log.Error(err, "failed to create a new instance of the cluster query builder")
+		return false, err
+	}
+
+	apiGroup1 := capdiscovery.Group("packageMetadateAPIQuery", "data.packaging.carvel.dev").WithVersions("v1alpha1").WithResource("packagemetadatas")
+	apiGroup2 := capdiscovery.Group("packageAPIQuery", "data.packaging.carvel.dev").WithVersions("v1alpha1").WithResource("packages")
+	apiGroup3 := capdiscovery.Group("packageRepositoryAPIQuery", "packaging.carvel.dev").WithVersions("v1alpha1").WithResource("packagerepositories")
+	apiGroup4 := capdiscovery.Group("packageInstallAPIQuery", "packaging.carvel.dev").WithVersions("v1alpha1").WithResource("packageinstalls")
```

Comment:
question: Is there a way we could discover resources without hardcoding the version? A use case I could think of is:

* I have v1alpha1 carvel with my current cluster
* I upgrade my cluster which has v1beta1 carvel now
* My current package plugin will break because there will be no `v1alpha1` anymore

Reply:
not sure how to discover the API version but I believe the upgrade case will also affect secretGen API version

https://github.com/vmware-tanzu/tanzu-framework/blob/main/cmd/cli/plugin/secret/main.go

cc @maralavi

Reply:
By looking at this [code](https://github.com/vmware-tanzu/tanzu-framework/blob/978797e5abd67ee1a8b0e1a087340f87e4ae3dd4/pkg/v1/sdk/capabilities/discovery/cluster_gvr.go#L51), I dont think `WithVersions` is required. Could you help test it out ?

Reply:
yeah I've deleted `WithVersions` and it will fall into this error:
https://github.com/vmware-tanzu/tanzu-framework/blob/978797e5abd67ee1a8b0e1a087340f87e4ae3dd4/pkg/v1/sdk/capabilities/discovery/cluster_gvr.go#L53
`
Error: failed to check for the availability of 'packaging.carvel.dev/v1alpha1' API: cannot check for group and resource existence without version info; use WithVersion method`

Reply:
added constants to represent API name and version instead of hardcoding